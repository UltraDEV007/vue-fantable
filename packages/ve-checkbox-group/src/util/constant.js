// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
