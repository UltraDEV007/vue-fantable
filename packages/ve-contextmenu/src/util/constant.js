// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
