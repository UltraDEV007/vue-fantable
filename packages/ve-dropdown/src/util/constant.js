// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
