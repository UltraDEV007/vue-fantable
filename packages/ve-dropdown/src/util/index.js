/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
