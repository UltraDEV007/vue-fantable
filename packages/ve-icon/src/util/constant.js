// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
