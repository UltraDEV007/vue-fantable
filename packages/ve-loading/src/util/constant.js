// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
