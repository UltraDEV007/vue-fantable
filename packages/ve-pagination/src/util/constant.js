// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
