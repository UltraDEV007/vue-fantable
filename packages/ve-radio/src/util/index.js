import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
