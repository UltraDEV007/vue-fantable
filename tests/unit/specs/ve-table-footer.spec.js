import { mount } from "@vue/test-utils";
import veTable from "@/ve-table";
import { later } from "../util";

describe("veTable footer", () => {
    const TABLE_DATA = [
        {
            rowKey: 1001,
            name: "John",
            date: "1900-05-20",
            hobby: "coding",
            address: "No.1 Century Avenue, Shanghai",
        },
        {
            rowKey: 1002,
            name: "Dickerson",
            date: "1910-06-20",
            hobby: "coding",
            address: "No.1 Century Avenue, Beijing",
        },
        {
            rowKey: 1003,
            name: "Larsen",
            date: "2000-07-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Chongqing",
        },
        {
            rowKey: 1004,
            name: "Geneva",
            date: "2010-08-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Xiamen",
        },
        {
            rowKey: 1005,
            name: "Jami",
            date: "2020-09-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shenzhen",
        },
    ];

    const FOOTER_DATA = [
        {
            rowkey: 0,
            name: "平均值",
            date: 213,
            hobby: 355,
            address: 189,
        },
        {
            rowkey: 1,
            name: "汇总值",
            date: 1780,
            hobby: 890,
            address: 2988,
        },
    ];

    const COLUMNS = [
        {
            field: "name",
            key: "a",
            title: "Name",
            align: "center",
            fixed: "left",
            renderFooterCell: ({ row, column, rowIndex }, h) => {
                return (
                    <span class="text-bold" style="">
                        {row.name}
                    </span>
                );
            },
        },
        {
            field: "date",
            key: "b",
            title: "Date",
            align: "left",
        },
        {
            field: "hobby",
            key: "c",
            title: "Hobby",
            align: "center",
        },
        { field: "address", key: "d", title: "Address", align: "left" },
    ];

    it("render", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                footerData: FOOTER_DATA,
                rowKeyFieldName: "rowKey",
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("custom cell content", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                footerData: FOOTER_DATA,
                rowKeyFieldName: "rowKey",
            },
        });

        expect(
            wrapper
                .findAll(".ve-table-footer-tr")
                .at(0)
                .findAll(".ve-table-footer-td .text-bold")
                .exists(),
        ).toBe(true);
    });

    it("custom cell style", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                footerData: FOOTER_DATA,
                rowKeyFieldName: "rowKey",
                cellStyleOption: {
                    footerCellClass: ({ row, column, rowIndex }) => {
                        if (column.field === "address") {
                            return "table-footer-cell-class1";
                        }

                        if (column.field === "date" && rowIndex === 1) {
                            return "table-footer-cell-class2";
                        }
                    },
                },
            },
        });

        expect(
            wrapper
                .findAll(".ve-table-footer-tr")
                .at(0)
                .findAll(".ve-table-footer-td")
                .at(3)
                .find(".table-footer-cell-class1")
                .exists(),
        ).toBe(true);

        expect(
            wrapper
                .findAll(".ve-table-footer-tr")
                .at(1)
                .findAll(".ve-table-footer-td")
                .at(1)
                .find(".table-footer-cell-class2")
                .exists(),
        ).toBe(true);
    });

    it("cell span", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                footerData: FOOTER_DATA,
                rowKeyFieldName: "rowKey",
                cellSpanOption: {
                    footerCellSpan: ({ row, column, rowIndex }) => {
                        if (rowIndex === 0) {
                            if (column.field === "date") {
                                return {
                                    rowspan: 1,
                                    colspan: 2,
                                };
                            }
                            // does not need to be rendered
                            else if (column.field === "hobby") {
                                return {
                                    rowspan: 0,
                                    colspan: 0,
                                };
                            }
                        }
                    },
                },
            },
        });

        const tdEl = wrapper
            .findAll(".ve-table-footer-tr")
            .at(0)
            .findAll(".ve-table-footer-td")
            .at(1);

        expect(tdEl.attributes("rowspan")).toBe("1");

        expect(tdEl.attributes("colspan")).toBe("2");
    });

    it("custom row events", () => {
        const mockClickFn = jest.fn();
        const mockDblclickFn = jest.fn();
        const mockContextmenuFn = jest.fn();
        const mockMouseenterFn = jest.fn();
        const mockMouseleaveFn = jest.fn();

        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                footerData: FOOTER_DATA,
                rowKeyFieldName: "rowKey",
                eventCustomOption: {
                    footerRowEvents: ({ row, rowIndex }) => {
                        return {
                            click: (event) => {
                                mockClickFn(row, rowIndex, event);
                            },
                            dblclick: (event) => {
                                mockDblclickFn(row, rowIndex, event);
                            },
                            contextmenu: (event) => {
                                mockContextmenuFn(row, rowIndex, event);
                            },
                            mouseenter: (event) => {
                                mockMouseenterFn(row, rowIndex, event);
                            },
                            mouseleave: (event) => {
                                mockMouseleaveFn(row, rowIndex, event);
                            },
                        };
                    },
                },
            },
        });

        const firstTrEl = wrapper.findAll(".ve-table-footer-tr").at(0);

        firstTrEl.trigger("click");
        expect(mockClickFn).toBeCalledWith(
            FOOTER_DATA[0],
            0,
            expect.any(Object),
        );

        firstTrEl.trigger("dblclick");
        expect(mockDblclickFn).toBeCalledWith(
            FOOTER_DATA[0],
            0,
            expect.any(Object),
        );

        firstTrEl.trigger("contextmenu");
        expect(mockContextmenuFn).toBeCalledWith(
            FOOTER_DATA[0],
            0,
            expect.any(Object),
        );

        firstTrEl.trigger("mouseenter");
        expect(mockMouseenterFn).toBeCalledWith(
            FOOTER_DATA[0],
            0,
            expect.any(Object),
        );

        firstTrEl.trigger("mouseleave");
        expect(mockMouseleaveFn).toBeCalledWith(
            FOOTER_DATA[0],
            0,
            expect.any(Object),
        );
    });

    it("custom cell events", () => {
        const mockClickFn = jest.fn();
        const mockDblclickFn = jest.fn();
        const mockContextmenuFn = jest.fn();
        const mockMouseenterFn = jest.fn();
        const mockMouseleaveFn = jest.fn();

        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                footerData: FOOTER_DATA,
                rowKeyFieldName: "rowKey",
                eventCustomOption: {
                    footerCellEvents: ({ row, column, rowIndex }) => {
                        return {
                            click: (event) => {
                                mockClickFn(row, column, rowIndex, event);
                            },
                            dblclick: (event) => {
                                mockDblclickFn(row, column, rowIndex, event);
                            },
                            contextmenu: (event) => {
                                mockContextmenuFn(row, column, rowIndex, event);
                            },
                            mouseenter: (event) => {
                                mockMouseenterFn(row, column, rowIndex, event);
                            },
                            mouseleave: (event) => {
                                mockMouseleaveFn(row, column, rowIndex, event);
                            },
                        };
                    },
                },
            },
        });

        const firstTrTdEl = wrapper
            .findAll(".ve-table-footer-tr")
            .at(0)
            .findAll(".ve-table-footer-td")
            .at(0);

        firstTrTdEl.trigger("click");
        expect(mockClickFn).toBeCalledWith(
            FOOTER_DATA[0],
            expect.any(Object),
            0,
            expect.any(Object),
        );

        firstTrTdEl.trigger("dblclick");
        expect(mockDblclickFn).toBeCalledWith(
            FOOTER_DATA[0],
            expect.any(Object),
            0,
            expect.any(Object),
        );

        firstTrTdEl.trigger("contextmenu");
        expect(mockContextmenuFn).toBeCalledWith(
            FOOTER_DATA[0],
            expect.any(Object),
            0,
            expect.any(Object),
        );

        firstTrTdEl.trigger("mouseenter");
        expect(mockMouseenterFn).toBeCalledWith(
            FOOTER_DATA[0],
            expect.any(Object),
            0,
            expect.any(Object),
        );

        firstTrTdEl.trigger("mouseleave");
        expect(mockMouseleaveFn).toBeCalledWith(
            FOOTER_DATA[0],
            expect.any(Object),
            0,
            expect.any(Object),
        );
    });

    it("footer not fixed", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                footerData: FOOTER_DATA,
                rowKeyFieldName: "rowKey",
                fixedFooter: false,
            },
        });

        expect(
            wrapper.findAll(".ve-table-fixed-footer .ve-table-footer").exists(),
        ).toBe(false);
    });

    it("column fixed", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: [
                    {
                        field: "name",
                        key: "a",
                        title: "Name",
                        align: "center",
                        fixed: "left",
                        width: 200,
                    },
                    {
                        field: "date",
                        key: "b",
                        title: "Date",
                        align: "left",
                    },
                    {
                        field: "hobby",
                        key: "c",
                        title: "Hobby",
                        align: "center",
                    },
                    {
                        field: "address",
                        key: "d",
                        title: "Address",
                        align: "left",
                        fixed: "right",
                        width: 200,
                    },
                ],
                tableData: TABLE_DATA,
                footerData: FOOTER_DATA,
                rowKeyFieldName: "rowKey",
            },
        });

        expect(
            wrapper.find(".ve-table-header-th.ve-table-fixed-left").exists(),
        ).toBe(true);

        expect(
            wrapper
                .find(
                    ".ve-table-body-td.ve-table-fixed-left.ve-table-last-left-fixed-column",
                )
                .exists(),
        ).toBe(true);

        expect(
            wrapper
                .find(
                    ".ve-table-body-td.ve-table-fixed-right.ve-table-first-right-fixed-column",
                )
                .exists(),
        ).toBe(true);
    });
});
