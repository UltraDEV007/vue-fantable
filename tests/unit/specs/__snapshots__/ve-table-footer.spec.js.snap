// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veTable footer render 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col>
            <col>
            <col>
            <col>
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-fixed-left fan-table-last-left-fixed-column" style="text-align: center; left: 0px; top: 0px;">Name</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Date</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">Hobby</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: left; top: 0px;">Address</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr row-key="1001" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td fan-table-fixed-left fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">John</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">1900-05-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shanghai</td>
            </tr>
            <tr row-key="1002" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td fan-table-fixed-left fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">Dickerson</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">1910-06-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Beijing</td>
            </tr>
            <tr row-key="1003" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td fan-table-fixed-left fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">Larsen</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2000-07-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Chongqing</td>
            </tr>
            <tr row-key="1004" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td fan-table-fixed-left fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">Geneva</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2010-08-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Xiamen</td>
            </tr>
            <tr row-key="1005" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td fan-table-fixed-left fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">Jami</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2020-09-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shenzhen</td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer">
            <tr class="fan-table-footer-tr">
              <td rowspan="1" colspan="1" class="fan-table-footer-td fan-table-fixed-left fan-table-last-left-fixed-column" style="text-align: center; left: 0px; bottom: 0px;"><span class="text-bold">平均值</span></td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;">213</td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: center; bottom: 0px;">355</td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;">189</td>
            </tr>
            <tr class="fan-table-footer-tr">
              <td rowspan="1" colspan="1" class="fan-table-footer-td fan-table-fixed-left fan-table-last-left-fixed-column" style="text-align: center; left: 0px; bottom: 0px;"><span class="text-bold">汇总值</span></td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;">1780</td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: center; bottom: 0px;">890</td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;">2988</td>
            </tr>
          </tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left"></div>
          <div class="fan-table-selection-middle"></div>
          <div class="fan-table-selection-fixed-right"></div>
        </div>
      </div>
    </div>
    <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
  </div>
</div>
`;
